recipe/patches/0002-Add-support-of-setting-CAPath-and-relevant-test.patch

```diff
@@ -0,0 +1,227 @@
+From bf652dcd47905ad1d23346bf4ee7192287dcd388 Mon Sep 17 00:00:00 2001
+From: Phoebus Mak <61957902+phoebusm@users.noreply.github.com>
+Date: Fri, 29 Sep 2023 17:32:47 +0100
+Subject: [PATCH] Add support of setting CAPath and relevant test (#4982)
+
+* Add support of setting CAPath and relevant test
+
+* Renaming macro and update when will it get define
+---
+ sdk/core/azure-core/CHANGELOG.md              |  3 ++
+ .../inc/azure/core/http/curl_transport.hpp    | 22 ++++++++++
+ sdk/core/azure-core/src/http/curl/curl.cpp    | 19 +++++++++
+ .../test/ut/curl_connection_pool_test.cpp     | 14 ++++---
+ .../azure-core/test/ut/curl_options_test.cpp  | 42 +++++++++++++++++++
+ 5 files changed, 94 insertions(+), 6 deletions(-)
+
+diff --git a/sdk/core/azure-core/CHANGELOG.md b/sdk/core/azure-core/CHANGELOG.md
+index 84051931ff..f7dc726aae 100644
+--- a/sdk/core/azure-core/CHANGELOG.md
++++ b/sdk/core/azure-core/CHANGELOG.md
+@@ -4,6 +4,9 @@
+ 
+ ### Features Added
+ 
++- [[#4983]](https://github.com/Azure/azure-sdk-for-cpp/issues/4983) Added support for setting `CURLOPT_CAPATH` libcurl option on Linux.
++
++
+ ### Breaking Changes
+ 
+ ### Bugs Fixed
+diff --git a/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp b/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
+index d0fb2b4e82..9e417afcfe 100644
+--- a/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
++++ b/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
+@@ -12,6 +12,14 @@
+ #include "azure/core/http/http.hpp"
+ #include "azure/core/http/policies/policy.hpp"
+ #include "azure/core/http/transport.hpp"
++#include "azure/core/platform.hpp"
++
++#if defined(AZ_PLATFORM_LINUX)
++#include <openssl/opensslv.h>
++#if OPENSSL_VERSION_NUMBER >= 0x00905100L
++#define _azure_SUPPORT_SETTING_CAPATH
++#endif // OPENSSL_VERSION_NUMBER >= 0x00905100L
++#endif // defined(AZ_PLATFORM_LINUX)
+ 
+ namespace Azure { namespace Core { namespace Http {
+   class CurlNetworkConnection;
+@@ -122,6 +130,20 @@ namespace Azure { namespace Core { namespace Http {
+      */
+     std::string CAInfo;
+ 
++#if defined(_azure_SUPPORT_SETTING_CAPATH)
++    /**
++     * @brief Path to a directory which holds PEM encoded file, containing the certificate
++     * authorities sent to libcurl handle directly.
++     *
++     * @remark The Azure SDK will not check if the path is valid or not.
++     *
++     * @remark The default is the built-in system specific path. More about this option:
++     * https://curl.se/libcurl/c/CURLOPT_CAPATH.html
++     *
++     */
++    std::string CAPath;
++#endif
```

Comment:
This introduces a new public API. Are we sure it can be backported?

Reply:
I am not entirely sure, indeed. Also a maintainer of the Azure SDK for C++ mentioned that the next release should happen sometime soon (see https://github.com/Azure/azure-sdk-for-cpp/pull/4982#issuecomment-1741187849). Probably we wait for it?

Reply:
That looks like a better choice.

Reply:
To hopefully clarify, we are planning to release this change as 1.11.0-beta.1 on Thursday, October 5th.
Please note, since this goes to a beta release, and the beta release is for the library already had stable releases, the further Betas do not go to the official vcpkg repo. Usually, the beta changes are baked for a month until they are released in stable release. If you are looking for this change to appear in the vcpkg official repo, you'll probably need to wait until, I'd say, November 10th.
Otherwise, this change will be available in our Beta vcpkg registry - https://github.com/Azure/azure-sdk-vcpkg-betas - it should happen on October 5th.

Reply:
Thanks for chiming in @antkmsft! BTW, when can we expect the next stable release of `azure-identity-cpp`?

Reply:
We are targeting the November timeframe for the next stable release of `azure-identity-cpp`, if not sooner.

Is there something particular you are hoping to use from there? We'd love to get some usage feedback on the beta release to help stabilize it.